src/main/java/com/example/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,28 +1,53 @@
 package com.example.onlinebookstore.service.impl;
 
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.CreateBookRequestDto;
+import com.example.onlinebookstore.mapper.BookMapper;
 import com.example.onlinebookstore.model.Book;
 import com.example.onlinebookstore.repository.BookRepository;
 import com.example.onlinebookstore.service.BookService;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto bookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toModel(bookRequestDto)));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto get(Long id) {
+        return bookMapper.toDto(bookRepository.findById(id).orElseThrow(() ->
+                new RuntimeException("There is not book in db by id %d"
+                .formatted(id))));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public void delete(Long id) {
+        bookRepository.deleteById(id);
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
+        if (!bookRepository.existsById(id)) {
```

Comment:
bad logic
First, you need to find book by Id.
Second, create mapping method like 

`void updateBook(CreateBookRequestDto requestDto, @MappingTarget Book book);`

so
```
bookMapper.updateBook(requestDto, book);
return bookMapper.toDto(bookRepository.save(book));
```

Reply:
fixed in previous PR